Forge/src/main/java/me/ferdz/placeableitems/client/network/ClientPacketHandler.java

```diff
@@ -0,0 +1,53 @@
+package me.ferdz.placeableitems.client.network;
+
+import java.util.function.Supplier;
+
+import me.ferdz.placeableitems.network.SUpdateFluidHolderPacket;
+import me.ferdz.placeableitems.network.handler.AnonymousPacketHandler;
+import me.ferdz.placeableitems.tileentity.FluidHolderTileEntity;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.LazyLoadBase;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.World;
+
+import net.minecraftforge.fml.network.NetworkEvent;
+
+/**
+ * Client-sided implementation of {@link AnonymousPacketHandler}
+ *
+ * @author Parker Hawke - Choco
+ */
+public final class ClientPacketHandler implements AnonymousPacketHandler {
```

Comment:
This whole packet handler I'm confused by. Why do you need this? If you're using TileEntities, shouldn't that be enough to update the model across the network?

Reply:
No. Without the packet handler, no client other than the one interacting with the tile entity would be aware of the changes. Each client needs this information in order to render it properly, therefore a packet is sent to all clients tracking the updated chunk when required. This is Forge's preferred method when updating the values of a tile entity.

This differs from the `StackHolderTileEntity` because the held ItemStack only needs to be known to the server. The client has no knowledge of the stack because its logic is performed entirely server side. This, however, is not the case for the `FluidHolderTileEntity` which requires client-sided rendering.

Reply:
I'll double check later but I have the impression that the StackHolderTileEntity did impact rendering. Maybe I'm confusing with some older versions though

Reply:
`StackHolderTileEntity` has no special rendering based on the contained ItemStack. You may be thinking of the saddle stand which has differing models based on its state but is not a stack holder.

Reply:
You're correct, I got confused with some earlier versions where rendering depended on TileEntity properties. [The ingots used to work that way in previous versions](https://github.com/Ferdzz/PlaceableItems/blob/master/1.12.2/src/main/java/me/ferdz/placeableitems/tileentity/TEIngot.java), along with potions, cookies & tipped arrows. With that solution I was able to modify the BlockState based on the TE state, what do you think?

Reply:
With that codebase being significantly out of date with regards to the changes Forge made handling its tile entities and client/server interactions, I think it's safer to send required information through a packet. This is still the most optimized and recommended approach to synchronizing tile entity data, especially for client-sided rendering (see the docs [here](https://mcforge.readthedocs.io/en/1.15.x/tileentities/tileentity/)).

Reply:
Okay. For the lifetime of this project I always tried to keep everything as simple as possible, but if that's what's required, then so be it. Would you mind providing some documentation that may explain how each parth kind of "assembles" together? PacketHandlers, Packets, etc

Reply:
I can understand the desire to keep the project as minimal as possible, but it also restricts the project from growing. The keybind PR awaiting approval of this one also uses a packet and will require a network regardless.

Unfortunately there's minimal documentation on the networking aspect of Forge. The best I can send you is the official docs https://mcforge.readthedocs.io/en/1.15.x/networking/simpleimpl/ which doesn't explain the use of my abstraction in the packet handler. The reason I abstract it out the way I do is because the network channel and packets are registered on both the client and server in order to recognize them properly on both ends. However, this leads to classloading issues if the server tries to load something declares as `@OnlyIn(Dist.CLIENT)`.

A good example of this exact application of abstraction can be seen in vanilla's codebase. Forge maps these to `INetHandler`. See its various implementations. It takes similar inspiration 😄

Reply:
I see. Let's keep it like this